client/components/draggable/index.jsx

```diff
@@ -0,0 +1,125 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import noop from 'lodash/noop';
+import omit from 'lodash/omit';
+
+export default React.createClass( {
+	displayName: 'Draggable',
+
+	propTypes: {
+		onDrag: React.PropTypes.func,
+		onStop: React.PropTypes.func,
+		width: React.PropTypes.number,
+		height: React.PropTypes.number,
+		x: React.PropTypes.number,
+		y: React.PropTypes.number,
+		controlled: React.PropTypes.bool,
+		bounds: React.PropTypes.shape( {
+			top: React.PropTypes.number,
+			left: React.PropTypes.number,
+			bottom: React.PropTypes.number,
+			right: React.PropTypes.number
+		} )
+	},
+
+	getDefaultProps() {
+		return {
+			onDrag: noop,
+			onStop: noop,
+			width: 0,
+			height: 0,
+			x: 0,
+			y: 0,
+			controlled: false,
+			bounds: null
+		};
+	},
+
+	getInitialState() {
+		return {
+			x: this.props.x,
+			y: this.props.y,
+			relativePos: null
+		};
+	},
+
+	componentWillReceiveProps( newProps ) {
+		if ( this.state.x !== newProps.x || this.state.y !== newProps.y ) {
+			this.setState( {
+				x: newProps.x,
+				y: newProps.y
+			} );
+		}
+	},
+
+	componentWillUnmount() {
+		this.removeListeners();
+	},
+
+	onMouseDown( event ) {
+		event.preventDefault();
+
+		document.addEventListener( 'mousemove', this.onMouseMove );
+		document.addEventListener( 'mouseup', this.onMouseUp );
+
+		this.setState( {
+			relativePos: {
+				x: event.pageX - this.state.x,
+				y: event.pageY - this.state.y
+			}
+		} );
+	},
+
+	onMouseMove( event ) {
+		window.requestAnimationFrame( () => {
```

Comment:
This will queue up all the mousemove events.

For a great example of how to call the update events independently of mousemove check out http://www.html5rocks.com/en/tutorials/speed/animations/#debouncing-mouse-events.